x/config/config.go

```diff
@@ -171,3 +171,59 @@ func NewPacketListener(transportConfig string) (transport.PacketListener, error)
 		return nil, fmt.Errorf("config scheme '%v' is not supported", url.Scheme)
 	}
 }
+
+func SanitizeConfig(transportConfig string) (string, error) {
+	// Do nothing if the config is empty
+	if transportConfig == "" {
+		return "", nil
+	}
+	// Split the string into parts
+	parts := strings.Split(transportConfig, "|")
+
+	// Iterate through each part
+	for i, part := range parts {
+		url, err := parseConfigPart(part)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse config part: %w", err)
+		}
+		// This can be extended to support other schemes that need sanitization
+		if strings.HasPrefix(part, "ss://") {
+			host := url.Host
+			if host == "" {
+				return "", errors.New("host not specified in ss:// config")
+			}
+			prefixStr := url.Query().Get("prefix")
+			if len(prefixStr) > 0 {
+				parts[i] = "ss://" + "[redacted]@" + host + "?prefix=" + prefixStr
+			} else {
+				parts[i] = "ss://" + "[redacted]@" + host
+			}
+		}
+	}
+
+	// Join the parts back into a string
+	return strings.Join(parts, "|"), nil
+}
+
+func GetHostnamesFromConfig(transportConfig string) ([]string, error) {
```

Comment:
@fortuna The idea here is to get all the hostnames in provided transports and run a separate DNS tests and include resolved IP addresses in the report for each hostname. This DNS resolution is independent of the resolution done by the `Dialer` before connecting and we will probably do this sparingly.